### PR TITLE
Add the to_axis_angle() function for quaternion.

### PR DESCRIPTION
**Pull request by Fr3nchK1ss**
Also change the dmd compiler version in .travis.yml.

### CODE REVIEW

File: gl3n/linalg.d [hunk 1]
```diff
@@ -2228,6 +2228,57 @@ struct Quaternion(type) {
...
+            ret.y = ret.z = 0;
+        }
+        else
+        {
+            ret.x = x / denominator;
+            ret.y = y / denominator;
+            ret.z = z / denominator;
+        }
+
+        //return ret.normalized * angle;
```
Comment (Dav1dde) [inline gl3n/linalg.d:2251]: Leftover?

File: gl3n/linalg.d [hunk 2]
```diff
@@ -2228,6 +2228,57 @@ struct Quaternion(type) {
...
+        return ret * angle;
+    }
+    unittest {
+        // See https://www.energid.com/resources/orientation-calculator
+
+        void testPair(quat q, vec3 v)
```
Comment (Dav1dde) [inline gl3n/linalg.d:2257]: Codestyle: `{` on same line

File: gl3n/linalg.d [hunk 3]
```diff
@@ -2228,6 +2228,57 @@ struct Quaternion(type) {
...
+        {
+            vec3 q2v = q.to_axis_angle();
+            assert( almost_equal(q2v.x, q2v.x) /// epsilon = 0.000001f
```
Comment (Dav1dde) [inline gl3n/linalg.d:2260]: Codestyle: Space

File: gl3n/linalg.d [hunk 4]
```diff
@@ -2228,6 +2228,57 @@ struct Quaternion(type) {
...
     }
+    
+    /// Returns the quaternion as a vec3 (axis / angle representation).
+    vec3 to_axis_angle() {
+        vec3 ret;
+        quat this_normalized = this.normalized();
+        real angle = 2 * acos(this_normalized.w);
+        real denominator = sqrt(1.0 - (this_normalized.w)^^2);
+
+        if ( denominator < 0.0001) // avoid divide by 0
```
Comment (Dav1dde) [inline gl3n/linalg.d:2239]: Code Style would be:

```
if(denominator < 0.0001) {
```

No spaces and `{` on same line (same for `else` below)

File: .travis.yml [hunk 5]
```diff
@@ -4,7 +4,6 @@ d:
   - dmd
   - gdc
   - ldc
-  - dmd-2.066.1
-  - dmd-2.070.0-b1
+  - dmd-2.093.1
   - ldc-0.14.0
```
Comment (Dav1dde) [inline .travis.yml:8]: Should we update LDC and GDC as well?

**Review by Dav1dde — COMMENTED** [review]
Sorry I am that nitpicky with the Code-Style, if you feel like that is unnecessary, I won't object, since you're basically taking over the project and keeping it alive.

The only things non-codestyle:
* GDC/LDC Versions
* `almost_equal`

File: .travis.yml (hunk 5 above)
Comment (Fr3nchK1ss) [reply .travis.yml:8]: I can not test gdc and ldc, so I don't have version infos for an update. But yes I can delete the lines with hard-coded version of those compilers.
And no problem with the coding style. I totally agree on keeping consistency.

**Comment by Fr3nchK1ss** [comment]
Here we go, I pushed the changes.

**Pull request merged**